Port completness react query

olethanh opened the pull request.

beygorghor commented on hat/assets/js/apps/Iaso/domains/completeness/index.js:
```diff
@@ -1,76 +1,49 @@
+const Completeness = ({ params }) => {
+    const { formatMessage } = useSafeIntl();
+    const dispatch = useDispatch();
+    const { data = [], isFetching } = useQuery(['completeness'], () =>
```
i would use a requests.js file in each domain to group them. We have only one request for now but it's to be consistent/

beygorghor commented:
Also action column is sortable and sometimes looks useless
<img width="1595" alt="Screenshot 2021-09-06 at 13 43 25" src="https://user-images.githubusercontent.com/12494624/132212466-f33e34d5-bd05-4926-a477-40aa75f315cc.png">

olethanh commented:
Rebased on master, added back test that was removed by error. 
Modified to use useSnackMutation (and move it from polio to iaso), added a useSnackQuery. Both of theses function are just like the regulard useMutation and useQuery but handle the snackbar directly for us. which make the code a bit more readable

quang-le commented on hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessListComponent.js:
```diff
@@ -11,12 +11,7 @@ import { groupCompletenessData } from '../utils';
-    onGenerateDerivedInstances,
-}) {
+function CompletenessListComponent({ classes, completenessList }) {
     const [activePeriodType, setActivePeriodType] =
```
small nitpick: I would change to an arrow function as it seems to be the standard way to make function components

quang-le commented on hat/assets/js/apps/Iaso/domains/completeness/components/CompletenessPeriodComponent.js:
```diff
@@ -47,38 +57,49 @@ const CompletenessPeriodComponent = ({
         const derived = form.generate_derived;
-        onGenerateDerivedInstances({ periods, derived });
+        mutation.mutate({ periods, derived });
     };
```
I would rename `mutation.mutate` to make it more expressive

quang-le reviewed (commented):
It sure trims a lot of fat
